netci.groovy

```diff
@@ -644,525 +666,565 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
                                     return
                                 }
                                 break
-                            case 'ilrt':
-                                // The ilrt build isn't necessary except for os's in the cross list or Windows_NT (native OS runs)
-                                if (os != 'Windows_NT' && !(os in Constants.crossList)) {
+                            case 'arm':
+                                // Only Ubuntu cross implemented
+                                if (os != 'Ubuntu') {
                                     return
                                 }
-                                // Only x64 for now
-                                if (architecture != 'x64') {
+                                break
+                            case 'x86':
+                                // Skip non-windows
+                                if (os != 'Windows_NT') {
                                     return
                                 }
                                 break
-                            case 'default':
-                                // Nothing skipped
+                            case 'x64':
+                                // Everything implemented
                                 break
                             default:
-                                println("Unknown scenario: ${scenario}")
+                                println("Unknown architecture: ${architecture}")
                                 assert false
                                 break
                         }
-                    }
-                
-                    // Calculate names
-                    def lowerConfiguration = configuration.toLowerCase()
-                    def jobName = getJobName(configuration, architecture, os, scenario, isBuildOnly)
-                    
-                    // Create the new job
-                    def newJob = job(Utilities.getFullJobName(project, jobName, isPR)) {}
 
-                    setMachineAffinity(newJob, os, architecture)
+                        // We only do crossgen for default and pri1, so skip other scenarios.
+                        // We also only do Checked and x64, so skip configurations and architectures that don't match those
+                        if (doCrossgen) {
+                            if (scenario != 'default' && scenario != 'pri1') {
+                                return
+                            }
+                            if (configuration != 'Checked') {
+                                return
+                            }
+                            if (architecture != 'x64') {
+                                return
+                            }
+                            //We only need to build with crossgen on Windows, so we skip non-windows crossgen scenarios
+                            if  (os != 'Windows_NT') {
```

Comment:
Yes, this check is only for the build phase. R2R jobs will run for all OS's that we currently run Pri1 tests for (Windows, OSX, Ubuntu, CentOS)